lib/image-stream/gzip.js

```diff
@@ -29,46 +26,40 @@ const fs = Bluebird.promisifyAll(require('fs'));
 const ISIZE_LENGTH = 4;
 
 /**
- * @summary Get a gzip file uncompressed size
+ * @summary Get the estimated uncompressed size of a gzip file
  * @function
  * @public
  *
  * @description
  * This function determines the uncompressed size of the gzip file
- * by reading its `ISIZE`. The specification clarifies that this
- * value is just an estimation.
+ * by reading its `ISIZE` field at the end of the file. The specification
+ * clarifies that this value is just an estimation.
  *
- * @param {String} file - path to gzip file
+ * @param {Object} options - options
+ * @param {Number} options.size - file size
+ * @param {Function} options.read - read function (count, offset)
  * @fulfil {Number} - uncompressed size
  * @returns {Promise}
  *
  * @example
- * gzip.getUncompressedSize('path/to/file.gz').then((uncompressedSize) => {
+ * const fd = fs.openSync('path/to/image', 'r');
+ *
+ * gzip.getUncompressedSize({
+ *   size: fs.statSync('path/to/image.gz').size,
+ *   read: (count, offset) => {
+ *     const buffer = Buffer.alloc(count);
+ *     fs.readSync(fd, buffer, 0, count, offset);
+ *     return buffer;
```

Comment:
Looks like this should be returning a `Promise`, given `options.read().then()` on line 62